web_external/js/views/body/ImagesView/ImageWall.js

```diff
@@ -25,52 +21,49 @@ isic.views.ImagesViewSubViews.ImageWall = isic.View.extend({
         }
     },
     render: _.debounce(function () {
-        var self = this;
-
         d3.select(this.el)
           .selectAll('img')
           .remove();
 
         var sel = d3.select(this.el)
             .selectAll('img')
-            .data(this.model.images.pluck('_id'));
+            .data(this.model.images.map(function (image) {
+                return {
+                    id: image.id,
+                    name: image.get('name')
+                };
+            }));
 
         sel.enter()
             .append('img')
             .classed('thumb', true);
 
         sel.attr('src', function (d) {
-            return girder.apiRoot + '/image/' + d + '/thumbnail?width=' + imageSize;
+            return girder.apiRoot + '/image/' + d.id + '/thumbnail?width=' + imageSize;
         })
             .attr('height', imageSize * 0.75)
             .attr('width', imageSize)
             .attr('data-toggle', 'tooltip')
             .attr('data-placement', 'auto')
             .attr('data-viewport', '#isic-images-imageWall')
             .classed('selected', _.bind(function (d) {
-                return d === this.image.id;
+                return d.id === this.image.id;
             }, this))
             .on('click', _.bind(function (d) {
                 this.clearTooltips();
                 if (d3.event.shiftKey) {
-                    var image = new isic.models.ImageModel({
-                        _id: d
-                    });
                     new isic.views.ImageFullscreenWidget({ // eslint-disable-line no-new
                         el: $('#g-dialog-container'),
-                        model: image,
+                        model: this.model.images.findWhere({_id: d.id}),
```

Comment:
Why not just stuff the model into the data binding in the hash in line 31ff. above? That would eliminate this quadratic time lookup approach.